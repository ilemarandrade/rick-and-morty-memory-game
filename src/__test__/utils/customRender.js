import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import CharactersProvider from "../../providers/CharactersProvider";

const AllTheProviders = ({ children }) => {
  return (
    <BrowserRouter>
      <CharactersProvider>{children}</CharactersProvider>
    </BrowserRouter>
  );
};

const customRender = (ui, { ...options } = {}) =>
  render(ui, {
    wrapper: ({ children }) => <AllTheProviders children={children} />,
    ...options,
  });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
