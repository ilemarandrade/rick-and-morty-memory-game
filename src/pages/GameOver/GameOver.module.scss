@use "../../root/App.scss";

.root {
  margin: auto;
  width: min-content;

  h2 {
    font-size: 32px;
  }
}

.containerButtons {
  margin-top: 48px;
  flex-direction: column;
  gap: 25px;

  * {
    width: 100%;
  }
}
@media (min-width: #{App.$sm}) {
  .containerButtons {
    margin-top: 72px;
    flex-direction: row;
  }

  .root {
    h2 {
      font-size: 48px;
    }
  }
}
