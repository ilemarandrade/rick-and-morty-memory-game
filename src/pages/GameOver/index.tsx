import { NavLink } from "react-router-dom";
import Button from "../../components/Button";
import routes from "../../constants/routes";
import { useGameControlState } from "../../contexts/GameControlContext";
import classes from "./GameOver.module.scss";

const GameOver = () => {
  const { turns } = useGameControlState();

  return (
    <div className={classes.root}>
      <h2 className="mb-1 font-primary">Â¡Felicitaciones!</h2>
      <p className="mb-1-5">{`Terminaste el juego con ${turns} turnos`}</p>
      <div className={`d-flex jc-space-between ${classes.containerButtons}`}>
        <Button label="Repetir" component={NavLink} to={routes.PLAY} />
        <Button
          label="Inicio"
          component={NavLink}
          to={routes.HOME}
          color="secondary"
        />
      </div>
    </div>
  );
};

export default GameOver;
