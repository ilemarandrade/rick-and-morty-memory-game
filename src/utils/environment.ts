const env = {
  API_BASE_URL: process.env.REACT_APP_API_URL,
  APP_ENV: process.env.REACT_APP_APP_ENV,
};

export default env;
