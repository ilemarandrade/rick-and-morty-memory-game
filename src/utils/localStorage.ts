export const KEY_SHOW_INTRUCTIONS = "show-intructions-key";

export const getKey = (key: string) => localStorage.getItem(key);

export const setKey = ({ key, value }: { key: string; value: string }) =>
  localStorage.setItem(key, value);
