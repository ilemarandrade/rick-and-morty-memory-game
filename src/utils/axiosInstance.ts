import axios from "axios";
import environment from "./environment";

const instance = axios.create({
  baseURL: environment.API_BASE_URL,
  headers: {
    Accept: "application/json",
    "Content-Type": "application/json",
  },
});

export default instance;
