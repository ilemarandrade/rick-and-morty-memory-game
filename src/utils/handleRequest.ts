import { AxiosRequestConfig, Method } from "axios";
import axiosInstance from "./axiosInstance";
interface IhandleRequest<T> {
  method: Method;
  url: string;
  config?: AxiosRequestConfig;
  onSuccess: (res: { data: T }) => void;
  onError: (res: { data: T }) => void;
}

const handleRequest = <T>({
  method,
  url,
  config,
  onSuccess,
  onError,
}: IhandleRequest<T>) => {
  axiosInstance({
    method,
    url,
    ...config,
  })
    .then((res) => {
      onSuccess(res);
    })
    .catch((err) => {
      onError(err);
    });
};

export default handleRequest;
