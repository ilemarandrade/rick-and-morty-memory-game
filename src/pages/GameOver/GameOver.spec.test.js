import { render, screen, waitFor } from "../../__test__/utils/customRender";
import GameOver from "./index";

describe("GameOver", () => {
  beforeEach(() => {
    jest.resetModules();
  });
  it("should render without errors", async () => {
    const { container } = render(<GameOver />);
    // snapshot testing
    await waitFor(() => {
      screen.getByText("Â¡Felicitaciones!");
    });
    expect(container).toMatchSnapshot();
  });
});
