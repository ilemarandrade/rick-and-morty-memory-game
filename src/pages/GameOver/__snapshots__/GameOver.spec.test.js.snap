// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOver should render without errors 1`] = `
<div>
  <div
    class="root"
  >
    <h2
      class="mb-1 font-primary"
    >
      Â¡Felicitaciones!
    </h2>
    <p
      class="mb-1-5"
    >
      Terminaste el juego con 0 turnos
    </p>
    <div
      class="d-flex jc-space-between containerButtons"
    >
      <a
        class="button-base button-color-primary"
        color="primary"
        href="/play"
      >
        Repetir
      </a>
      <a
        aria-current="page"
        class="button-base button-color-secondary active"
        color="secondary"
        href="/"
      >
        Inicio
      </a>
    </div>
  </div>
</div>
`;
